oneview/data_source_server_profile_template.go

```diff
@@ -624,28 +1052,97 @@ func dataSourceServerProfileTemplateRead(d *schema.ResourceData, meta interface{
 	if spt.Bios != nil {
 		biosOptions := make([]map[string]interface{}, 0, 1)
 		biosOptions = append(biosOptions, map[string]interface{}{
+			"compliance_control":  spt.Bios.ComplianceControl,
 			"manage_bios":         spt.Bios.ManageBios,
 			"overridden_settings": overriddenSettings,
 		})
 
 		d.Set("bios_option", biosOptions)
 	}
 
+	firmwareOption := make([]map[string]interface{}, 0, 1)
+	firmwareOption = append(firmwareOption, map[string]interface{}{
+		"compliance_control":       spt.Firmware.ComplianceControl,
+		"firmware_activation_type": spt.Firmware.FirmwareActivationType,
+		"firmware_baseline_uri":    spt.Firmware.FirmwareBaselineUri.String(),
+		"firmware_install_type":    spt.Firmware.FirmwareInstallType,
+		"force_install_firmware":   spt.Firmware.ForceInstallFirmware,
+		"manage_firmware":          spt.Firmware.ManageFirmware,
+	})
+	d.Set("firmware", firmwareOption)
+
+	osCustomAttributes := make([]map[string]interface{}, 0, len(spt.OSDeploymentSettings.OSCustomAttributes))
+	for i := 0; i < len(spt.OSDeploymentSettings.OSCustomAttributes); i++ {
```

Comment:
it will be good to put this also inside If condition.

Reply:
done